packages/fast-form-generator-react/src/form/form-item.children.tsx

```diff
@@ -155,19 +182,23 @@ class FormItemChildren extends React.Component<IFormItemChildrenProps & IManaged
      * Click event for adding a component
      */
     private onAddComponent(item: any): void {
-        const currentChildren: JSX.Element = get(this.props.data, "children");
-        const currentChildrenArray: JSX.Element[] = this.getCurrentChildArray(currentChildren);
-        const dataLocation: string = isRootLocation(this.props.dataLocation) ? `children` : `${this.props.dataLocation}.children`;
+        const currentChildrenArray: React.ReactNode[] = this.getCurrentChildArray(this.props.data);
 
         if (typeof item === "object") {
             this.props.onChange(
-                dataLocation,
-                this.getReactComponents(currentChildrenArray, item)
+                this.props.dataLocation,
+                this.getReactComponents(currentChildrenArray, item),
+                undefined,
+                undefined,
+                true
             );
         } else if (typeof item === "string") {
             this.props.onChange(
-                dataLocation,
-                this.getStringComponents(currentChildrenArray, item)
+                this.props.dataLocation,
```

Comment:
What are your thoughts on making `onChange` receive an object instead of a big list of arguments? I think that would be more readable and prevent you from needing to pass `undefined` all the time.

If that seems reasonable but not in scope for this change I can file a separate issue.

Reply:
It could be a config object sure, let's create another issue to consolidate.